### DISCUSSION TITLE
Pull request file filter should have "Select None" option

### DISCUSSION THREAD

**Discussion by MarkB-WetaInteractive** [community/community#29204] [Pull Requests]
I'm working on a Unity project, where many binary or otherwise uninteresting files need to be committed, but I don't want to review any of them as part of my PR. There's a helpful button to select all once you have deselected any file types:

![image](https://user-images.githubusercontent.com/109932147/184242206-a3a30d36-5ff3-4159-a962-bdd95362e4f0.png)

If that screen space was taken up with a button that said "Deselect all file types" when _all_ file types were selected, it would considerably increase the efficiency of this UX; otherwise it's a lot of quite precise clicking required every time you navigate away from your PR.

**Comment by KevinGelkingSviper** [comment]
I very much agree. And ideally, it should include per-repository pre-filters.
For each PR I have to do 11-16 clicks to untick everything other than .cs files and those CS files then usually end up not being loaded because Git prefers to load the diffs for junk files instead that require no review.

E.g. here, after 11 clicks to untick everything besides .cs all the CS file diffs are not loaded because they werent at the top of the queue - so I have to click all 35 of those manually as well.
The PR functionality of Github is unfortunately extremely lacking for any project that isn't purely code.

<img width="656" alt="chrome_kw1s7uRjPz" src="https://user-images.githubusercontent.com/103119563/189616861-6ce8726c-1366-41b3-9e6e-2fb14fffb208.png">

**Comment by eexit** [comment]
Hello, 

Any update on this?

Thanks.

**Comment by Vishwajeet-keni** [comment]
Hi @MarkB-WetaInteractive and @KevinGelkingSviper!

A 'Deselect all' toggle would definitely be a huge quality-of-life improvement for game development repos. In the meantime, here are two tricks that solve both the repetitive clicking and the 'Load diff' budget problem for Unity PRs:*

1. Collapse Unity junk files automatically via .gitattributes
You can instruct GitHub to treat .meta, .asset, .prefab, etc. as generated files so they stay collapsed by default and don't block .cs diffs from loading:

Add this to your repo's .gitattributes file:

Code snippet
*.meta linguist-generated=true
*.asset linguist-generated=true
*.prefab linguist-generated=true
*.unity linguist-generated=true
Once added, GitHub will collapse these by default in all future PRs, freeing up the automatic diff loader for your actual C# code!

2. URL File Filter Bookmark
You can also instantly filter a PR view down to just C# files by appending ?file-filters%5B%5D=.cs to the end of your PR's Files changed tab URL.

**Comment by EricBatlle** [comment]
I know it's an old post, but I made an extension for Chrome that may help you: 
[Chrome Web Store](https://chromewebstore.google.com/detail/github-pr-file-filters/hjhdneklbjnidhhjgobijhfdpkjinncl)
[GitHub Repository](https://github.com/EricBatlle/Github-PR-File-Filters)
